packages/expect-puppeteer/src/index.js

```diff
@@ -53,12 +53,24 @@ function expectPage(page) {
   return expectation
 }
 
+function checkConstructorName(type) {
```

Comment:
I think checking constructor name is not enough, it can have conflict, we should try a more reliable way to check it. Unfortunately Puppeteer does not expose its internal classes...

Reply:
I am open to suggestions on a better way to check for this. What you do think would work as better identifier?

I figured checking the constructor name was a viable alternative to checking the methods attached to the object. As an example, `Page` has `$$`, and ElementHandle also has [`$$`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#elementhandleselector-1). The thought process behind using their constructor names was avoiding identification of Puppeteer objects via APIs that may be more generic than expected, and could get deprecated.

Reply:
Checking `$$` method + constructor seems enough for me.

I understand your concern but I prefer to keep it safe if a user try to make assertion on his proper `Page` class:

```js
class Page {}

expect(new Page()).toBeInstanceOf(Page)
```

Checking `$$` reduces conflict changes. The better would be to have a reference on Puppeteer Page class but we haven't yet.